server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java

```diff
@@ -93,15 +96,43 @@ public void getSegmentFiles(
         ActionListener<GetSegmentFilesResponse> listener
     ) {
         try {
-            indexShard.syncSegmentsFromRemoteSegmentStore(false, true, false);
-            listener.onResponse(new GetSegmentFilesResponse(Collections.emptyList()));
+            logger.trace("Downloading segments files from remote store {}", filesToFetch);
+            FilterDirectory remoteStoreDirectory = (FilterDirectory) indexShard.remoteStore().directory();
+            FilterDirectory byteSizeCachingStoreDirectory = (FilterDirectory) remoteStoreDirectory.getDelegate();
+            RemoteSegmentStoreDirectory remoteSegmentStoreDirectory = (RemoteSegmentStoreDirectory) byteSizeCachingStoreDirectory
+                .getDelegate();
+            RemoteSegmentMetadata remoteSegmentMetadata = remoteSegmentStoreDirectory.init();
+            List<StoreFileMetadata> downloadedSegments = new ArrayList<>();
+            if (remoteSegmentMetadata != null) {
+                try {
+                    indexShard.store().incRef();
+                    indexShard.remoteStore().incRef();
+                    final Directory storeDirectory = indexShard.store().directory();
+                    String segmentNFile = null;
+                    for (StoreFileMetadata fileMetadata : filesToFetch) {
+                        String file = fileMetadata.name();
+                        storeDirectory.copyFrom(remoteSegmentStoreDirectory, file, file, IOContext.DEFAULT);
+                        downloadedSegments.add(fileMetadata);
+                        if (file.startsWith(IndexFileNames.SEGMENTS)) {
```

Comment:
I accidentally deleted my older comment and can't seem to recover now :-(

Anyways, since we're honouring the diff calculated by `SegmentReplicationTarget` now, which would skip the `segment_N` files([link](https://github.com/opensearch-project/OpenSearch/blob/e3542011cd9584c354405ba6cbf9a31ced7bc5ce/server/src/main/java/org/opensearch/index/store/Store.java#L395)),  the `segment_N` files shouldn't ideally be downloaded after this change. That was the intention of asking for assertion here. Also we can  remove the check for single segmentInfoSnapshot file as well.

Reply:
I see @ankitkala. Thanks for clarifying this. I added assert to not have `segment_N` file and removed logic for `segment_N` handling.

Reply:
Yes you are right @ankitkala. We need to include the `segment_N` file for remote store as we are not performing commits on replica. Added back logic to download segment_N file as well.